### ISSUE TITLE
Rate limiting

### ISSUE THREAD

**Issue by haarp** [keepassxreboot/keepassxc#5983]
## Summary
Integrations are able to quickly consume the entire password database. Maybe I didn't look closely enough, but I did not find any rate limitations. A rogue extension would be able to dump the entire database in no time.

I propose a configurable rate limitation of e.g. one entry per 5 seconds, or maybe even exponential. It's rare that you need many passwords in very quick succession, and configuration allows fine-tuning. It would give the user notice (through notifications) and the ability to stop the process.

## Examples
A rogue extension somehow manages to replace KeepassXC-Browser, or connect to KeepassXC. It then proceeds to dump passwords. Without rate-limiting, all db contents are now in the attacker's hands. With rate-limiting, the user sees password request notifications every 5 seconds. The user can now decide to lock the DB or quit KeepassXC. A few db entries may have leaked, but the biggest disaster is hopefully averted.

## Context
It's a possible attack vector that's very hard or impossible to fully prevent, but quite easy to limit. Why not? :)

Thanks a lot!

**Comment by droidmonkey** [comment]
I tend to agree on this. It's not perfect security but it's better than no limiting. Certainly doesn't prevent the "rogue extension" from patiently waiting 5 seconds per request. That still allows a dump within a very short time frame. All this withstanding, you do need to hijack the connection key to begin with and also guess which urls to retrieve credentials for.

**Comment by haarp** [comment]
> Certainly doesn't prevent the "rogue extension" from patiently waiting 5 seconds per request. That still allows a dump within a very short time frame.

True. The idea is that the user notices to be able to manually intervene. A rate limit with the purpose of preventing dumps would make it very difficult to find a balance between protection and pure annoyance.

> you do need to hijack the connection key to begin with

Can't fully trust browsers to keep that secure. Who knows what can happen in those 1M+ LOC behemoths.

> and also guess which urls to retrieve credentials for.

Not hard to just try a list of commonly used ones. Facebook, Twitter, Paypal, Github, etc.
